src/parser/template/parameters.ts

```diff
@@ -1,155 +1,35 @@
-import { parseNumber } from '../private/types';
-import { schema } from '../schema';
+import { CfnParameterProps } from 'aws-cdk-lib';
+import {
+  assertBoolean,
+  assertListOfForm,
+  assertNumber,
+  assertObject,
+  assertString,
+} from '../private/types';
+import { ifField } from './expression';
 
-export type ContextValue = string | string[] | symbol;
+export interface TemplateParameter extends CfnParameterProps {}
 
-export class TemplateParameters {
-  constructor(private readonly parameters: Record<string, schema.Parameter>) {}
-
-  public get required(): Record<string, schema.Parameter> {
-    return Object.fromEntries(
-      Object.entries(this.parameters).filter(
-        ([_, param]) => param.Default == null
-      )
-    );
-  }
-
-  public has(name: string) {
-    return !!this.parameters[name];
-  }
-
-  public parse(name: string, value: string) {
-    const param = this.parameters[name];
-    if (!param) {
-      throw new Error(`Unknown parameter: ${name}`);
-    }
-    return parseParamValue(name, param, value);
-  }
-}
-
-function parseParamValue(name: string, param: schema.Parameter, value: string) {
```

Comment:
All of this appears to be for local resolution of parameter values. We don't need that here.